src/EnablerExtension.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace SilverStripe\LoginForms;
+
+use SilverStripe\View\SSViewer;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\Security\Security;
+use SilverStripe\Core\Config\Config;
+
+/**
+ * Applies to the {@see Security} controller in order to detect requests for actions related to
+ * the log in process or other such credential management (such as the forgot password flow).
+ * This is in order to replace the set {@see SSViewer} theme list with a conrolled set in order
+ * to always show a consistent interface that relates more to the CMS than the website it is
+ * loading on.
+ * Particular actions can be set to be ignored by inlcuding them in the `excluded_actions` list
+ * defined in yml _config for this class. By default all allowed actions on the Security controller
+ * excepting `index`, `logout`, and `ping` will have the new theme set applied.
+ */
+class EnablerExtension extends DataExtension
+{
+    public function beforeCallActionHandler()
+    {
+        $config = Config::inst();
+        /** @var Security */
```

Comment:
If this doesn't mention a variable it's not actually doing anything. If you want to type hint the property you have to do it with a property level doc block in the class definition, or you can alias `$owner = $this->owner` and type hint `$owner`, but you can't typehint `$this->owner` inline

Reply:
This is quelling phpstan complaints that `DataObject` does not have a function `getAction` (because somewhere up the hierarchy it must be defined that `$this->owner` is such).

Come to think of it though, have I used `DataExtension`? I should use `Extension`.

Reply:
![image](https://user-images.githubusercontent.com/778003/56011122-49d9af00-5d3a-11e9-8f30-106f25a14340.png)

Reply:
This _controller targeted_ extension now uses `Extension`

Reply:
That's cool, but my comment was that your PHPDoc is invalid without a variable name in it

Reply:
It should be `/** @var Security $owner */` with `$owner = $this->getOwner()` if you want it to conform to PSR-19 (still not accepted). It's really an issue with the way extensions and PHPDoc works though, ideally getOwner() would annotate its return type correctly.

Reply:
Oh I see, thanks for clarifying.